### PR TITLE
chore: add desp-checker

### PR DESCRIPTION
**Pull request by meifans**
Move deps checker code and IT test to Fx-core

### CODE REVIEW

**Review by LongOddCode — APPROVED** [review]
LGTM

File: packages/fx-core/tests/common/deps-checker/utils/dotnet.ts [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+import {
+  DotnetChecker,
+  DotnetVersion,
+} from "../../../../src/common/deps-checker/internal/dotnetChecker";
+
+const find = require("find-process");
+
+tmp.setGracefulCleanup();
+
+export const dotnetConfigPath = path.join(os.homedir(), "." + ConfigFolderName, "dotnet.json");
```
Comment (swatDong) [inline packages/fx-core/tests/common/deps-checker/utils/dotnet.ts:23]: Is this Unit Test or Integration Test?
If Unit Test, do not use home folder since this will break existing toolkit.
If Integration Test, separate these tests from existing core unit tests, as well as github action.

File: packages/fx-core/package.json [hunk 2]
```diff
@@ -30,7 +30,7 @@
     "test:core-tools": "nyc mocha \"tests/core/tools.test.ts\"",
     "test:apiv2": "nyc mocha \"tests/plugins/resource/apiv2/*.test.ts\"",
     "test:solutionv3": "nyc mocha \"tests/plugins/solutionv3/*.test.ts\"",
-    "test:env-checker": "nyc mocha \"tests/common/deps-checker/*.test.ts\"",
+    "test:env-checker": "npx mocha \"tests/common/deps-checker/**/*.test.ts\"",
```
Comment (swatDong) [inline packages/fx-core/package.json]: why this change?

File: packages/fx-core/src/common/deps-checker/checkerFactory.ts [hunk 3]
```diff
@@ -0,0 +1,36 @@
+// Copyright (c) Microsoft Corporation.
```
Comment (swatDong) [inline packages/fx-core/src/common/deps-checker/checkerFactory.ts:1]: Adjust code owner for deps-checker, as well as tests.

File: packages/fx-core/src/common/deps-checker/internal/dotnetChecker.ts [hunk 4]
```diff
@@ -0,0 +1,534 @@
...
+import { DepsTelemetry } from "../depsTelemetry";
+import { DepsInfo, DepsChecker } from "../depsChecker";
+import { Messages } from "../constant/message";
+
+const execFile = util.promisify(child_process.execFile);
+
+export enum DotnetVersion {
+  v21 = "2.1",
+  v31 = "3.1",
+  v50 = "5.0",
```
Comment (swatDong) [inline packages/fx-core/src/common/deps-checker/internal/dotnetChecker.ts:28]: Make sure you are moving the latest code. E.g., we already have .NET 6.0 support.

File: packages/fx-core/package.json (hunk 2 above)
Comment (meifans) [reply packages/fx-core/package.json]: nyc will terminate IT cases by coverage check issue. You can see this [job](https://github.com/OfficeDev/TeamsFx/runs/4614820710?check_suite_focus=true) to learn more about.

File: packages/fx-core/src/common/deps-checker/internal/dotnetChecker.ts (hunk 4 above)
Comment (meifans) [reply packages/fx-core/src/common/deps-checker/internal/dotnetChecker.ts:28]: ok

File: packages/fx-core/tests/common/deps-checker/utils/dotnet.ts (hunk 1 above)
Comment (meifans) [reply packages/fx-core/tests/common/deps-checker/utils/dotnet.ts:23]: Ts files that match "*.test.ts " is UT. `dotnet.ts` is a Util used in `dotnet.it.ts`.

File: packages/fx-core/src/common/deps-checker/checkerFactory.ts (hunk 3 above)
Comment (meifans) [reply packages/fx-core/src/common/deps-checker/checkerFactory.ts:1]: Added

**Review by swatDong — APPROVED** [review]
LGTM

**Review by a1exwang — APPROVED** [review]
LGTM

**Review by LongOddCode — APPROVED** [review]
